Client/Frontend/Home/PanelDataObservers.swift

```diff
@@ -49,21 +49,31 @@ class ActivityStreamDataObserver: DataObserver {
     }
     
     func invalidate(highlights: Bool) {
+        guard !profile.isShutdown else {
+            return
+        }
+
         self.delegate?.willInvalidateDataSources()
 
         let notify = {
             self.delegate?.didInvalidateDataSources()
         }
-        
+
         let invalidateTopSites: () -> Success = {
             self.profile.history.setTopSitesNeedsInvalidation()
             return self.profile.history.updateTopSitesCacheIfInvalidated() >>> succeed
```

Comment:
This code suffers the same problem: `uTSCIF` ends up running a `DELETE`, then touching prefs, then running an `INSERT`, then touching prefs.